pandas/core/algorithms.py

```diff
@@ -415,33 +417,40 @@ def isin(comps, values):
     comps = com._values_from_object(comps)
 
     comps, dtype, _ = _ensure_data(comps)
-    values, _, _ = _ensure_data(values, dtype=dtype)
+
+    is_time_like = lambda x: (is_datetime_or_timedelta_dtype(x)
```

Comment:
i already indicate that this is not the path forward

this is much slower than the existing

you need to keep it along the current structure

and post benchmarks

Reply:
@jreback Thanks - have removed the flags `int_flg` / `float_flg` as suggested per your earlier review and posted the benchmarks after each commit. 
Re-pasting the benchmarks after the last commit
>ASV results on algorithms after 82386c3 are similar to the most recent one 5416711
```
       before           after         ratio
     [537b65cb]       [82386c31]
-     2.36±0.02ms      1.92±0.02ms     0.82  algorithms.Hashing.time_series_timedeltas

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
```
Also, could not get around writing a custom function `is_time_like` - had tried it in 
9fca52c 

Please feel free to close this PR - since I think I am not sure I understand what is the expectation in terms of maintaining the existing structure

Reply:
there are many benchmarks

the point is that this code is highly sensistive to changes and requires a lot of benchmark running to avoid regressions